doc/docs/FAQ.md

```diff
@@ -190,9 +192,37 @@ There are at least two possible reasons due to using: (1) a `material_function`
+
+### How do I create a focused beam with a Gaussian envelope?
+
+A focused beam with a Gaussian envelope can be created using the amplitude function (`amp_func`) of the [`Source`](Python_User_Interface.md#source) object. Examples are provided for [Python](https://github.com/NanoComp/meep/blob/master/python/examples/gaussian-beam.py) and [Scheme](https://github.com/NanoComp/meep/blob/master/scheme/examples/gaussian-beam.ctl). Four snapshots of the resulting field profile generated using this script for different values of the beam width (`sigma`) and rotation angle (`tilt_angle`) are shown in the following image:
+
+<center>
+![](images/gaussian_beam.png)
+</center>
+
+The [numerical aperture (NA)](https://en.wikipedia.org/wiki/Gaussian_beam#Beam_divergence) of a Gaussian beam of width w (2*`sigma` from the example script) and vacuum wavelength λ in a medium of index n is n⋅sin(λ/(πnw)).
+
+Note: in this example, the beam waist is at the source position (i.e., top center of the cell). If you want the beam waist to be at a position other than the position of the source, you need to adjust the *phase* of the beam accordingly. If you assume you have a Gaussian beam profile with zero phase at some plane y=y0, then you can work out the beam profile (including phase) at any other plane y=y1 by taking the Fourier transform and looking at the propagation of each planewave component, and then inverse Fourier transforming. In this way, you can work out the desired source profile at any plane y=y1 to get a Gaussian beam waist at y=y0.
+
+### How do I compute the effective index of an eigenmode of a lossy waveguide?
+
+To compute the [effective index](https://www.rp-photonics.com/effective_refractive_index.html), you will need to first compute the *complex* ω (the loss in time) for a *real* β (the propagation constant) and then convert this quantity into a loss in space (*complex* β at a *real* ω) by dividing by the group velocity v<sub>g</sub>. This procedure is described in more detail below.
+
+To obtain the loss in time, you make your computational cell a cross-section of your waveguide (i.e. 2d for a waveguide with constant cross-section), and set Bloch-periodic boundary conditions via the `k_point` input variable &mdash; this specifies your (real) β. You then treat it exactly the same as a [resonant-cavity problem](Python_Tutorials/Resonant_Modes_and_Transmission_in_a_Waveguide_Cavity.md#resonant-modes): you excite the system with a short pulse source, monitor the field at some point, and then analyze the result with [`Harminv`](Python_User_Interface.md#harminv); all of which is done if you call `run_kpoints`. This will give you the complex ω at the given β, where the imaginary part is the loss rate in time. Note: the loss in a uniform waveguide, with no absorption or disorder, is zero, even in the discretized system.
+
```

Comment:
Am I missing something, or does this FAQ never answer the question?  At the end, the effective index is usually defined as β/ω.

All the business with the group velocity is only if you need the *complex* effective index.  If you just want the real effective index (e.g. for a lossless waveguide), you can skip it.  Of course, in that case you could also use MPB.

Reply:
@jball1982: this was a bug. Should be fixed in #2499.

Reply:
@stevengj: this FAQ specifically addresses the case of *lossy* waveguides for which the answer provided would seem sufficient. Anyhow, for completeness, I have updated this FAQ in #2500 to also describe the case of lossless waveguides.

Reply:
@oskooi Thanks Ardavan, I appreciate your help. Despite the documentation on [readthedocs](https://meep.readthedocs.io/), I'm still having some trouble connecting the dots between C++ and Python in some cases. I am going to dig a little more and perhaps create a discussion on here soon with some questions.